account_bank_statement_import_online/models/online_bank_statement_provider.py

```diff
@@ -315,11 +348,24 @@ def _update_statement_balance_values(
             )
 
     @api.multi
-    def _get_filtered_lines(self, lines_data):
+    def _get_filtered_lines(
+        self,
+        lines_data,
+        statement_date_since,
+        statement_date_until
+    ):
         """Use unique import id to prevent duplicate lines."""
         AccountBankStatementLine = self.env["account.bank.statement.line"]
         filtered_lines = []
         for line_values in lines_data:
+            # Do not include transactions for dates not in statement period.
+            date = self._get_utc_line_date(line_values)
+            date = date.astimezone(utc).replace(tzinfo=None)
+            if date < statement_date_since:
+                continue
+            if date >= statement_date_until:
+                continue
```

Comment:
Looking at this again - isn't this dangerous, though? Example:

1. We have loaded stuff from day A on statement A
2. We are now in day B, but we fetch still some transactions from day A also, that are unknown in our system
3. With this new code, we don't load those, because our statement starts in day B and so we skip loading these transactions, and they're not in the system at all

Reply:
You are mistaken. See above.